go/vt/sqlparser/sql.y

```diff
@@ -602,19 +607,56 @@ base_select:
   }
 
 base_select_no_cte:
-  SELECT comment_opt cache_opt distinct_opt sql_calc_found_rows_opt straight_join_opt select_expression_list where_expression_opt group_by_opt having_opt window_opt
+  SELECT comment_opt cache_opt distinct_opt sql_calc_found_rows_opt straight_join_opt select_expression_list into_opt from_opt where_expression_opt group_by_opt having_opt window_opt into_opt
```

Comment:
Nice job combining these two rules into one with the new `from_opt` rule. I like that! 👍 

Digging into the rules a little more... it looks like `base_select_no_cte` is used as a subcomponent of the `base_select` rule, which is used by `select_statement`. It's in `select_statement` that the `order_by_opt` and `limit_opt` rules are used, so that actually seems like where we want to add `into_opt` for the end of the statement – otherwise, we couldn't parse valid forms such as `select id from mytable order by id desc limit 1 into @myvar`. Try adding a test case for that and see if we can shift around to support that.